lib/DiscoveryStreamFeed.jsm

```diff
@@ -286,18 +319,22 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
    * @param {RefreshAllOptions} options
    */
   async refreshAll(options = {}) {
-    const dispatch = options.updateOpenTabs ?
+    const {updateOpenTabs, isStartup} = options;
+    const dispatch = updateOpenTabs ?
       action => this.store.dispatch(ac.BroadcastToContent(action)) :
       this.store.dispatch;
 
-    await this.loadLayout(dispatch);
+    const layout = await this.loadLayout(dispatch, isStartup);
     await this.loadComponentFeeds(dispatch);
     await this.loadSpocs(dispatch);
+    await this._maybeUpdateLayout(isStartup, layout);
+
     this.loaded = true;
   }
 
-  async enable() {
-    await this.refreshAll({updateOpenTabs: true});
+  async enable(options = {}) {
+    const {isStartup} = options;
```

Comment:
I don't think you need this. I think we can always assume the `refreshAll` inside `enable` is always functionally a startup.

Reply:
The other case of a pref change can be treated as a startup too.

I'm only aware of three cases where we call enable. Init, pref change, system tick.